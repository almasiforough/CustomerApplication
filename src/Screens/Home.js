import React from 'react'
import { Grid } from '@material-ui/core'
import useStyles from './styles/Home.styles'
import CustomerTable from '../Components/CustomerTable'

export default function Home() {
  const classes = useStyles()
  return (
    <Grid className={classes.rootGrid}>
      <CustomerTable />
    </Grid>
  )
}
