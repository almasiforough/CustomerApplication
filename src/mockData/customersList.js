export const customersListMock = [
  {
    id: '6',
    lastName: 'Almasi',
    dateOfBirth: '1986-07-03',
    firstName: 'Forough'
  },
  {
    id: '7',
    lastName: 'beckham',
    dateOfBirth: '2020-11-17',
    firstName: 'david'
  },
  {
    id: '8',
    lastName: 'Beckham',
    dateOfBirth: '2014-06-17',
    firstName: 'Victoria'
  }
]
