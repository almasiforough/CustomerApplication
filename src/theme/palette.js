export default {
  primary: {
    main: '#37474f',
    light: '#62727b',
    dark: '#102027'
  },
  secondary: {
    main: '#b71c1c',
    light: '#f05545',
    dark: '#7f0000'
  }
  // text: {
  //   primary: '#7d8591',
  //   secondary: '#25d9da'
  // },
  // background: {
  //   main: '#f8fdff',
  //   white: '#fff',
  //   cityItem: '#fffbfb'
  // },
  // shadaow: {
  //   main: '#1b207626'
  // },
  // border: {
  //   main: '#d8d8d8',
  //   input: '#eaeaea'
  // }
}
