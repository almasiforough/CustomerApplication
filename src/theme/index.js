import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles'
import palette from './palette'

const theme = createMuiTheme({
  palette,
  typography: {
    fontFamily: 'Nunito',
    button: {
      lineHeight: 2.75
    }
  }
})
export default responsiveFontSizes(theme)
