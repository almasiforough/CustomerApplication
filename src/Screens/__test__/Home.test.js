import React from 'react'
import { create, act } from 'react-test-renderer'
import { ThemeProvider } from '@material-ui/core'
import { BrowserRouter } from 'react-router-dom'
import Home from '../Home'
import theme from '../../theme'
import { customersListMock } from '../../mockData/customersList'

describe('Homepage contains list of all customers', () => {
  test('take a snapshot from Home Component', async () => {
    const mockJsonPromise = Promise.resolve(customersListMock)
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
      status: 200
    })
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise)
    let homeComponent
    await act(async () => {
      homeComponent = create(
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <Home />
          </ThemeProvider>
        </BrowserRouter>
      )
    })

    expect(homeComponent.toJSON()).toMatchSnapshot()
  })

  test('every customers must be shown at first', async () => {
    const mockJsonPromise = Promise.resolve(customersListMock)
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
      status: 200
    })
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise)
    let homeComponent
    await act(async () => {
      homeComponent = create(
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <Home />
          </ThemeProvider>
        </BrowserRouter>
      )
    })

    const tableRows = homeComponent.root.findAllByType('tr')

    // There are three customers in mockData, plus on `tr` in table Header would be 4 `tr`s
    expect(tableRows.length).toBe(4)
  })

  test('check to see if searching by a typo still works.', async () => {
    const mockJsonPromise = Promise.resolve(customersListMock)
    const mockFetchPromise = Promise.resolve({
      json: () => mockJsonPromise,
      status: 200
    })
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise)
    let homeComponent
    await act(async () => {
      homeComponent = create(
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <Home />
          </ThemeProvider>
        </BrowserRouter>
      )
    })

    const searchInput = homeComponent.root.findByType('input')
    act(() => {
      // there's a customer named `forough`
      searchInput.props.onChange({ target: { value: 'frough' } })
    })

    const tableRows = homeComponent.root.findAllByType('tr')

    expect(tableRows.length).toBe(2)
  })
})
