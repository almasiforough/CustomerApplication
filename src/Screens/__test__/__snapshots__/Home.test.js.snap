// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage contains list of all customers take a snapshot from Home Component 1`] = `
<div
  className="MuiGrid-root makeStyles-rootGrid-1"
>
  <div
    className="MuiGrid-root makeStyles-rootGrid-2"
  >
    <div />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root makeStyles-rootGrid-8 MuiGrid-item"
      >
        <div
          className="MuiGrid-root makeStyles-searchGrid-10 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-searchNameInput-9"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Search
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Search in Customers first name or last name"
                required={false}
                type="text"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-13"
                >
                  <span>
                    Search
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root makeStyles-tableContainer-5 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeaderCell-4"
              scope="col"
            >
              First Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeaderCell-4"
              scope="col"
            >
              Last Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeaderCell-4"
              scope="col"
            >
              Date Of Birth
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            />
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            />
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Forough
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Almasi
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              1986-07-03
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <a
                href="/user/6"
                onClick={[Function]}
                user={
                  Object {
                    "dateOfBirth": "1986-07-03",
                    "firstName": "Forough",
                    "id": "6",
                    "lastName": "Almasi",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </a>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <button
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              Victoria
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              Beckham
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2014-06-17
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <a
                href="/user/8"
                onClick={[Function]}
                user={
                  Object {
                    "dateOfBirth": "2014-06-17",
                    "firstName": "Victoria",
                    "id": "8",
                    "lastName": "Beckham",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </a>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <button
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              role="cell"
              scope="row"
            >
              david
            </th>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              beckham
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              2020-11-17
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <a
                href="/user/7"
                onClick={[Function]}
                user={
                  Object {
                    "dateOfBirth": "2020-11-17",
                    "firstName": "david",
                    "id": "7",
                    "lastName": "beckham",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </a>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <button
                className="MuiButtonBase-root MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
